org.eclipse.jdt.core/s4jie2-tests/runner/org/eclipse/jdt/core/s4jie2TestSuite.java

```diff
@@ -233,9 +283,155 @@ public static void main(String[] args) throws IOException {
+				+ "	at Main.main(multiple_throw_conditions_first_satisfied_thrown.java:12)\n");
+		testCompileAndRun(true, "multiple_throw_conditions_second_satisfied_thrown", false, "", 
+				"Exception in thread \"main\" java.lang.ArithmeticException\n"
+				+ "	at Main.foo(multiple_throw_conditions_second_satisfied_thrown.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_second_satisfied_thrown.java:12)\n");
+		testCompileAndRun(true, "multiple_throw_conditions_none_satisfied_none_thrown", true, "", "");
+		testCompileAndRun(true, "multiple_throw_conditions_first_satisfied_none_thrown", false, "", 
+				"Exception in thread \"main\" java.lang.AssertionError: @throws condition holds but specified exception type not thrown\n"
+				+ "	at Main.foo$post(multiple_throw_conditions_first_satisfied_none_thrown.java:4)\n"
+				+ "	at Main.foo(multiple_throw_conditions_first_satisfied_none_thrown.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_first_satisfied_none_thrown.java:12)\n");
+		testCompileAndRun(true, "multiple_throw_conditions_second_satisfied_none_thrown", false, "", 
+				"Exception in thread \"main\" java.lang.AssertionError: @throws condition holds but specified exception type not thrown\n"
+				+ "	at Main.foo$post(multiple_throw_conditions_second_satisfied_none_thrown.java:5)\n"
+				+ "	at Main.foo(multiple_throw_conditions_second_satisfied_none_thrown.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_second_satisfied_none_thrown.java:12)\n");
+		testPartOfStringCompileAndRun(true, "multiple_throw_conditions_none_satisfied_first_thrown", false, "", 
+				"SEVERE: The thrown exception was not specified in the formal specification\n"
+				+ "Exception in thread \"main\" java.lang.IllegalArgumentException\n"
+				+ "	at Main.foo(multiple_throw_conditions_none_satisfied_first_thrown.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_none_satisfied_first_thrown.java:12)\n");
+		testPartOfStringCompileAndRun(true, "multiple_throw_conditions_none_satisfied_second_thrown", false, "",
+				"SEVERE: The thrown exception was not specified in the formal specification\n"
+				+ "Exception in thread \"main\" java.lang.ArithmeticException\n"
+				+ "	at Main.foo(multiple_throw_conditions_none_satisfied_second_thrown.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_none_satisfied_second_thrown.java:12)\n");
+		testCompileAndRun(true, "multiple_may_throw_conditions_none_satisfied_no_throw", true, "", "");
+		testCompileAndRun(true, "multiple_may_throw_conditions_first_satisfied_no_throw", true, "", "");
+		testCompileAndRun(true, "multiple_may_throw_conditions_second_satisfied_no_throw", true, "", "");
+		testCompileAndRun(true, "multiple_may_throw_conditions_satisfied_no_throw", true, "", "");
+		testPartOfStringCompileAndRun(true, "multiple_may_throw_conditions_none_satisfied_first_thrown", false, "", 
+				"SEVERE: @may_throw condition does not hold for the thrown exception type\n"
```

Comment:
In general, multiple @may_throw clauses might apply to a given exception, either if there are multiple clauses for the same exception type, or if one clause's exception type is a supertype of another clause's exception type. So perhaps it's best to always simply generate the generic error "The thrown exception was not specified in the formal specification"?